### ISSUE TITLE
Additional characters being returned with the horoscope message

### ISSUE THREAD

**Issue by mattchue** [tapaswenipathak/Horoscope-API#8]
Because this scrapes things from GaneshaSpeaks unofficially, I imagine this bug has been introduced by their doing.

The `horoscope` property of the result has an unnessicary `['` before the actual horoscope content. 

See [this](http://horoscope-api.herokuapp.com/horoscope/today/scorpio) request. The extra characters can be seen at the beginning of the `horoscope` value.

```
{
  "date": "17-02-2016", 
  "horoscope": "[' The company of like-minded people will keep you in cheery spirit throughout the day. The day promises to be lively and vibrant as you will engage in one activity after another. But Ganesha warns that being over enthusiastic may be detrimental to your health. You energy level will be at an all-time high as you enjoy soaring to new heights in the company of loved ones.", 
  "sunsign": "scorpio"
}
```

**Comment by ghost** [comment]
Any suggestions  to solve this?

**Comment by zak123** [comment]
What I did to solve it was just edit my code locally, its in Objective-C but its really basic, just seeing if the string starts with "[" and deleting everything up to the first letter

```
-(NSString*)formatHoroscopeText:(NSString *)horoscopeText {

    if ([horoscopeText hasPrefix:@"["]) {
        NSString *newStr = [horoscopeText substringFromIndex:3];

        return newStr;
    } else {
        return horoscopeText;

    }


}
```

It's not a perm fix and if it ever has a new issue with some other character prefixing strings I'll have to update the code again but I'm sure someone else can find a better solution

**Comment by skined90** [comment]
Hi,

There are few issues with the API.
First, on request for some period, horoscope value starts with the " [' " character. 
Second things is, on knowmore API call, weird json is returned:

`{
  "Type": "Air-Cardinal-Positive", 
  "lord": "\\n']", 
  "lucky_day": "['Venus : Blue, Green", 
  "lucky_number": "6, 15, 24, 33, 42, 51", 
  "meaning_of_name": "The Scales", 
  "sanskrit_name": "Tula"
}`

Something is messed up.

**Comment by tapaswenipathak** [comment]
Are they still coming?

**Comment by StefanWerW** [comment]
@tapasweni-pathak Yes. If you try http://horoscope-api.herokuapp.com/horoscope/today/taurus , it returned:

> {
>   "date": "02-06-2016", 
>   "horoscope": "[' Ganesha urges you to be prepared for setbacks and disappointments today. Your efforts may end in being a waste of precious time and energy. The results you get are likely to be far below your expectations. Falling short of your target is likely to weigh you down with stress and anxiety. You can, however, ease and reduce your frustration by reexamining your expectations and lowering them down to a comfortable, achievable level.", 
>   "sunsign": "taurus"
> }

Still coming the `[` char.

**Comment by KarthikaReddy** [comment]
{
"date": "26-08-2016"
"horoscope": "[' Most of your time will be spent at the work place. You will excel in all your tasks today. Professional relationships will be marked by co-operation. You will have cordial relations with your colleagues. It is an auspicious and progressive day for business, says Ganesha."
"sunsign": "Leo"
}

[ starting char

**Comment by angien** [comment]
:( Still doing it.

**Comment by RameshRacharla** [comment]
Hi,

We can rectify like this:

String DATA="";
 if(DATA.contains("['"))
            {
                String[] str=DATA.split("'");
                String message = str[1];
                String name = " ";
                String result = message.replaceAll("\\\\u", name);
                String name1 ="Astroleger";
                String finalresult = result.replaceAll("Ganesha",name1);

                textView.setText("" + finalresult);

                return;
            }

**Comment by kamalverma91** [comment]
Hello 
Tapasweni 

This link is not working showing error 404 not found .
http://horoscope-api.herokuapp.com/knowmore/libra

**Comment by adarrrshc** [comment]
any workaround for java?

**Comment by akshayrupareliya0912** [comment]
https://www.ganeshaspeaks.com/horoscopes/yesterday-horoscope/sunsign

**Comment by akshayrupareliya0912** [comment]
it works
